rfd/0174-kube-rbac-bootstrap.md

```diff
@@ -0,0 +1,293 @@
+can save on the backend (DynamoDB has a limit of 400kb), therefore, if some users would have an unusually large amount of Kubernetes RBAC resources they 
+want to provision, they would need to split it into two KubeProvision resources.
+
+
+#### Technical details
+
+```protobuf
+import "teleport/header/v1/metadata.proto";
+
+// KubeProvision represents a Kubernetes resources that can be provisioned on the Kubernetes clusters.
+// This includes roles/role bindings and cluster roles/cluster role bindings.
+// For rationale behind this type, see the RFD 174.
+message KubeProvision {
+  // The kind of resource represented.
+  string kind = 1;
+  // Not populated for this resource type.
+  string sub_kind = 2;
+  // The version of the resource being represented.
+  string version = 3;
+  // Common metadata that all resources share.
+  teleport.header.v1.Metadata metadata = 4;
+  // The specific properties of kube provision.
+  KubeProvisionSpec spec = 5;
+}
+
+// KubeProvisionSpec is the spec for the kube provision message.
+message KubeProvisionSpec {
+  // cluster_roles contains definitions for ClusterRole Kubernetes resources to provision.
+  repeated ClusterRole cluster_roles = 1;
+
+  // cluster_role_bindings contains definitions for ClusterRoleBinding Kubernetes resources to provision.
+  repeated ClusterRoleBinding cluster_role_bindings = 2;
+
+  // roles contains definitions for Role Kubernetes resources to provision.
+  repeated Role roles = 3;
+
+  // role_bindings contains definitions for RoleBinding Kubernetes resources to provision.
+  repeated RoleBinding role_bindings = 4;
+}
+
+// KubeProvisionService is a service that provides methods to manage KubeProvisions.
+service KubeProvisionService {
+  // CreateKubeProvision creates a new KubeProvision.
+  rpc CreateKubeProvision(CreateKubeProvisionRequest) returns (KubeProvision);
+  
+  // GetKubeProvision gets a KubeProvision by name.
+  rpc GetKubeProvision(GetKubeProvisionRequest) returns (KubeProvision);
+  
+  // ListKubeProvisions returns a list of KubeProvisions. It supports pagination.
+  rpc ListKubeProvisions(ListKubeProvisionsRequest) returns (ListKubeProvisionsResponse);
+  
+  // UpdateKubeProvision updates an existing KubeProvision.
+  rpc UpdateKubeProvision(UpdateKubeProvisionRequest) returns (KubeProvision);
+  
+  // UpsertKubeProvision upserts a KubeProvision.
+  rpc UpsertKubeProvision(UpsertKubeProvisionRequest) returns (KubeProvision);
+  
+  // DeleteKubeProvision deletes a KubeProvision.
+  rpc DeleteKubeProvision(DeleteKubeProvisionRequest) returns (google.protobuf.Empty);
+  
+  // DeleteAllKubeProvisions removes all KubeProvisions.
+  rpc DeleteAllKubeProvisions(DeleteAllKubeProvisionsRequest) returns (google.protobuf.Empty);
+}
+```
+
+Every five minutes we will run a reconciliation loop that will compare the current state on Kubernetes clusters under
```

Comment:
Authorization verdicts happen once per request, so exec session will continue, since the connection already exists. That's standard Kubernetes behaviour, same as if someone would manually change RBAC.